Add a fetch latest script

bcomnes opened the pull request:
Next steps include adding some tests to assert that used methods remain the same.

bcomnes committed:
Add a fetch latest script

Next steps include adding some tests to assert that used methods remain the same.

emilbayes commented:
Here's something similar with core node modules (in reference to your comment about streaming hashing in core 😃)

```js
const stream = require('stream')
const fs = require('fs')
const crypto = require('crypto')
const https = require('https')

const FILE_PATH = '...'
const FILE_URL = 'https://raw.githubusercontent.com/WebAssembly/wabt/master/demo/libwabt.js'

var httpHash
var fileHash

stream.pipeline(
  fs.createReadStream(FILE_PATH),
  crypto.createHash('sha256').on('data', (hash) => { fileHash = hash }),
  onfilehash
)

function onfilehash (err) {
  if (err) throw err

  https.get(FILE_URL, onresponse)
    .on('error', (err) => { throw err })
}

function onresponse (res) {
  stream.pipeline(
    res,
    crypto.createHash('sha256').on('data', (hash) => { httpHash = hash }),
    onhttphash
  )
}

function onhttphash (err) {
  if (err) throw err

  process.exit(httpHash.compare(fileHash))
}

```

But I generally think this is a good way to detect when libwabt updates, maybe in a similar vein to https://github.com/sodium-friends/sodium-native-citgm which runs on a Travis cron job

bcomnes commented:
That looks better.  I'll switch it over when I have some end of day time.

bcomnes committed:
Improve update scripts

bcomnes committed:
Better script names

emilbayes commented:
Nice, thanks Bret!

bcomnes commented:
Fixd

bcomnes committed:
Add simple tests to ensure wat files compile and work

Not sure why the updated local.get/local.set syntax isn't supported yet, even in the updated libwabt.js.

bcomnes commented:
Added a slightly more complex example that tests some of the keywords that appear changed in https://github.com/WebAssembly/spec/tree/master/interpreter#s-expression-syntax (e.g. local_set/local.set etc).  

I also added some simple test which actually consume the generated wasm.  

I tried using the new keywords, even with the updated libwabt (to be applied separately) but they don't seem to work yet.

bcomnes commented:
Good to land? Or was there more we wanted to do here.
